src/main/java/inputcommands/AddDeadlineCommand.java

```diff
@@ -0,0 +1,83 @@
+package inputcommands;
+
+import snomexceptions.InvalidCommandDateFormatException;
+import snomexceptions.InvalidCommandException;
+import snomexceptions.InvalidCommandTaskDescException;
+import snomtasklist.TaskList;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+
+
+
+
+
+/**
+ * The AddDeadlineCommand implements the command of
+ * adding a task of type Deadline into tasklist.
+ */
+class AddDeadlineCommand extends Command {
+
+    protected AddDeadlineCommand(String desc) {
+        super(desc);
+    }
+
+
+
+    /**
+     * {@inheritDoc}
+     * In this subclass, the command will return enum
+     * of type DEADLINE
+     *
+     * @return the enum of type DEADLINE
+     */
+    @Override
+    public CmdType getType() {
+        return CmdType.DEADLINE;
+    }
+
+    /**
+     * {@inheritDoc}
+     * In this subclass, we implement date checks and task description checks
```

Comment:
Might be better to have indentation after `{@inheritDoc}` tag
```suggestion
     * {@inheritDoc}
     *     In this subclass, we implement date checks and task description checks
```